### PR TITLE
better telegram gif/sticker handling

### PR DESCRIPTION
**Pull request by Bubu**
Respects tg bridge info fields to loop and hide the video player
controls. If autoplay is disabled it plays on hover instead.

Flags can be seen i.e. here: https://github.com/mautrix/telegram/commit/64107fab177ac7a31f811fa5f2409230b020bcbe

---
Here's what your changelog entry will look like:

## ✨ Features
 * better telegram gif/sticker handling ([\#6642](https://github.com/matrix-org/matrix-react-sdk/pull/6642)). Contributed by [Bubu](https://github.com/Bubu).

### CODE REVIEW

File: src/components/views/messages/MVideoBody.tsx [hunk 1]
```diff
@@ -240,6 +240,8 @@ export default class MVideoBody extends React.PureComponent<IBodyProps, IState>
         let width = null;
         let poster = null;
         let preload = "metadata";
+        let loop = false;
+        let controls = true;
```
Comment (SimonBrandner) [inline src/components/views/messages/MVideoBody.tsx:244]:
```suggestion
        const loop = content.info["fi.mau.loop"];
        const controls = !content.info["fi.mau.hide_controls"];
```

Or similar feels more readable. 

Also, could you please add some comment explaining why is this necessary?

File: src/components/views/messages/MVideoBody.tsx [hunk 2]
```diff
@@ -259,13 +267,20 @@ export default class MVideoBody extends React.PureComponent<IBodyProps, IState>
...
                     autoPlay={autoplay}
                     height={height}
                     width={width}
                     poster={poster}
+                    onMouseOver={!autoplay && !controls ? event => (event.target as HTMLVideoElement).play() : () => { }}
+                    onMouseOut={!autoplay && !controls ?
+                        event => {
+                            (event.target as HTMLVideoElement).pause();
+                            (event.target as HTMLVideoElement).currentTime = 0;
+                        } : () => { }}
```
Comment (SimonBrandner) [inline src/components/views/messages/MVideoBody.tsx:283]: I think this would be easier to read if these were separate methods somewhere above

**Review by SimonBrandner — CHANGES_REQUESTED** [review]

**Comment by Bubu** [comment]
All valid points, I can add more explanation as well :thumbsup:

**Comment by turt2live** [comment]
Why is the Telegram bridge sending videos as stickers :( It should be sending sticker events.

**Comment by Bubu** [comment]
I think sticker events can only contain images atm?

**Comment by turt2live** [comment]
They indeed only handle images at the moment, but the whole point of Matrix is that things are extensible and backwards compatible with each other. A sticker would have an image fallback (which may very well be a still image for performance reasons), but also metadata to indicate that it can be rendered as a looped video.

Extensible Events in particular would help with this, but it's not necessarily needed here.

**Review by turt2live — CHANGES_REQUESTED** [review]
Clearing review request. We should support looping video stickers off of stickers, not video events.

**Comment by Bubu** [comment]
> Clearing review request. We should support looping video stickers off of stickers, not video events.

For stickers, maybe. Telegram also sends "gifs" as videos (they are just mp4s in the telegram world because gifs suck).Those should behave like "real" gifs for the user I think.

I won't have time to implement animated sticker events, so closing.
